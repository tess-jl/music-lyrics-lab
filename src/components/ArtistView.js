import React from 'react'; 
import PropTypes from 'prop-types';
import useArtist from '../hooks/useArtist';
import { Link } from 'react-router-dom';

const ArtistView = ({ handleArtistSelection }) => {

  const { releases } = useArtist();

  const mappedReleases = releases.map(({ id, title }) => {
    return (
      <>
        <li key={id}>
          <Link to={`/album/${id}`}>
            <img src={`http://coverartarchive.org/release/${id}/front`} />
            <h3>{title}</h3>
          </Link>
        </li>
      </>
    );
  });
  return (
    <ul>
      {mappedReleases}
    </ul>
  );
};

export default ArtistView;
