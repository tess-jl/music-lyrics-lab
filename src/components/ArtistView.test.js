import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import ArtistView from './ArtistView';

describe('AlbumView component', () => {
  it('renders an Artist View', () => {
    const wrapper = shallow(<MemoryRouter><ArtistView /></MemoryRouter>);
    expect(wrapper).toMatchSnapshot();
  });
});
