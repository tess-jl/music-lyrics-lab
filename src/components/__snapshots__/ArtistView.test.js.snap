// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumView component renders an Artist View 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "veoet2",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "veoet2",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ArtistView />
</Router>
`;
